content/productivity/pink-it/plugin.js

```diff
@@ -0,0 +1,529 @@
+}
+
+function createToggleButton(text, onClick, onOrOff=false) {
+	const button = {};
+   	button.element = document.createElement('button');
+	button.element.innerText = text;
+	button.onOrOff = false;
+	button.set = function(b) {
+		button.onOrOff = b;
+		if (button.onOrOff) { // on
+			button.element.style.background='#CCC';
+			button.element.style.color='#000000';
+		} else { // off
+			button.element.style.background='#333333';
+			button.element.style.color='#FFF';
+		}
+	}
+	button.set(onOrOff);
+   	button.element.addEventListener('click', ()=>{
+		button.set(!button.onOrOff);
+		onClick(button.onOrOff);
+	});
+	return button;
+}
+
+function FullButtonLine(buttons) {
+	let table = document.createElement("table");
+	table.style.textAlign = "center";
+	table.style.width = "100%";
+	table.style.tableLayout = "fixed"; // to make all buttons the same width
+	for (let i=0; i < buttons.length; ++i) {
+		let b = buttons[i];
+		let td = document.createElement("td");
+		let ele = b.element ? b.element : b;
+		ele.style.border =  "1px solid white";
+		ele.style.width = "100%";
+		td.style.width = "100%"; // to make all buttons the same width
+		td.style.padding = "5px";
+		if (i !== 0) td.style.paddingLeft = "0px";
+		td.append(ele);
+		table.append(td);
+	}
+	return table;
+}
+
+function createTableHeader(table, strArr) {
+	var tr = document.createElement('tr');
+	for (var str of strArr) {
+		var th = document.createElement('th');
+		th.innerText = str;
+		th.style.border = "1px solid white";
+		tr.appendChild(th);
+	}
+	table.appendChild(tr);
+}
+
+function addAsTd(tr, text, width=null, color=null, center=true) {
+	var td = document.createElement('td');
+	td.style.border = "1px solid white";
+	td.innerHTML = text;
+	if (width) td.width = width+"px";
+	if (color) td.style.color = color;
+	if (center) td.style["text-align"] = "center";
+	tr.appendChild(td);
+	return td;
+}
+
+function intToStrHexColor(num) {
+	var color = num.toString(16).toUpperCase();
+	if (color.length < 2) color = "0"+color;
+	return color;
+}
+function percentToStrHexColor(percent) {
+	return intToStrHexColor(parseInt(percent*255));
+}
+function getPlanetRingRadius(planet, multiplier=1.3, min=minDrawPlanetRadius) {
+	const viewport = ui.getViewport();
+	let radius = viewport.worldToCanvasDist(ui.getRadiusOfPlanetLevel(planet.planetLevel));
+	radius *= multiplier;
+	if (radius < min) radius = min;
+	return radius;
+}
+function drawRingAnimationOnPlanet(ctx, planet, level, color, reverse=false) {
+	const viewport = ui.getViewport();
+	const { x: planetX, y: planetY } = viewport.worldToCanvasCoords(planet.location.coords);
+
+	let radius = getPlanetRingRadius(planet);
+
+	let animationDuration = 1500;
+	animationDuration *= (1 - level*0.1);
+	
+	var millis = Date.now();
+
+	for (var i=0; i<=level; ++i) {
+		drawRingAnimation(ctx, planetX, planetY, radius, millis, animationDuration, color, reverse);
+		millis += animationDuration / (level+1);
+	}
```

Comment:
```suggestion
	const radius = getPlanetRingRadius(planet);
	const animationDuration = 1500 * (1 - level * 0.1);
	
	let millis = Date.now();
	for (let  i = 0; i <= level; ++i) {
		drawRingAnimation(ctx, planetX, planetY, radius, millis, animationDuration, color, reverse);
		millis += animationDuration / (level+1);
	}
```

1. both `radius` and `animationDuration` are constants that never changed
2. Don't mind the `++i` still the `i++` could also be used.
3. rather use `let` over `var` keyword.